src/app/query/stacks/sip010/sip010-tokens.hooks.ts

```diff
@@ -0,0 +1,115 @@
+import { useMemo } from 'react';
+
+import type { FtMetadataResponse } from '@hirosystems/token-metadata-api-client';
+import type { Sip010CryptoAssetInfo } from '@leather-wallet/models';
+import BigNumber from 'bignumber.js';
+
+import { createMoney } from '@shared/models/money.model';
+import { isDefined, isUndefined } from '@shared/utils';
+
+import { pullContractIdFromIdentity } from '@app/common/utils';
+import {
+  useAlexCurrencyPriceAsMarketData,
+  useAlexSwappableAssets,
+} from '@app/query/common/alex-sdk/alex-sdk.hooks';
+import { type Sip010CryptoAsset, cryptoAssetFactory } from '@app/query/common/crypto-asset.model';
+import { useCurrentStacksAccountAddress } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { getAssetStringParts } from '@app/ui/utils/get-asset-string-parts';
+
+import { useStacksAccountBalanceFungibleTokens } from '../balance/stacks-ft-balances.hooks';
+import { useStacksAccountFungibleTokenMetadata } from '../token-metadata/fungible-tokens/fungible-token-metadata.hooks';
+import { isFtAsset } from '../token-metadata/token-metadata.utils';
+import { type Sip010CryptoAssetFilter, filterSip010CryptoAssets } from './sip010-tokens.utils';
+
+export function isTransferableStacksFungibleTokenAsset(asset: Partial<FtMetadataResponse>) {
+  return !isUndefined(asset.decimals) && !isUndefined(asset.name) && !isUndefined(asset.symbol);
+}
+
+function createSip010CryptoAssetInfo(
+  contractId: string,
+  key: string,
+  token: FtMetadataResponse
+): Sip010CryptoAssetInfo {
+  const { assetName, contractName } = getAssetStringParts(key);
+
+  return {
+    canTransfer: isTransferableStacksFungibleTokenAsset(token),
+    contractId,
+    contractName,
+    decimals: token.decimals ?? 0,
+    hasMemo: isTransferableStacksFungibleTokenAsset(token),
+    imageCanonicalUri: token.image_canonical_uri ?? '',
+    name: token.name ? token.name : assetName,
+    symbol: token.symbol ?? '',
+  };
+}
+
+const sip010CryptoAssetBalancePlaceholder = { availableBalance: createMoney(0, '') };
+
+const sip010CryptoAssetPlaceholder = cryptoAssetFactory<Sip010CryptoAsset>({
+  balance: sip010CryptoAssetBalancePlaceholder,
+  chain: 'stacks',
+  info: createSip010CryptoAssetInfo('', '', { sender_address: '', tx_id: '' }),
+  marketData: null,
+  type: 'sip-010',
+});
```

Comment:
Why is this needed, and what might we be able to do to avoid needing a dummy object?

Reply:
I feel like I always need something as a default with react query bc the query can be undefined? I always seem to run into this so maybe I'm missing something with how I setup the query/hooks? If I don't do this, later I have to use conditional props with hooks trying to pass the asset around.

Reply:
As in it's `undefined` because the query hasn't loaded yet? Better to pass around `Sip010CryptoAsset | null` than a  fake `Sip010CryptoAsset`.

Can you highlight an example case to look at where this is a problem?

Reply:
Feel free to merge this separately, but reckon it's part of the same concern and done as part of this effort. 

> It would be hard to pass an undefined asset thru all those hooks

Agree, that'd be a horrible solution as well. I'm hopefully there's another way where we just handle the null case properly in one place, and only ever call those methods when the value is defined?

Reply:
Happy to have a stab as well if that'd help?

Reply:
All good, let me refactor some things. 👍